platform.md

```diff
@@ -157,6 +157,18 @@ The stack ID:
 - MUST only contain numbers, letters, and the characters `.`, `/`, and `-`.
 - SHOULD use reverse domain name notation to avoid name collisions.
 
+### Stack Validation
+
+During the prepare phase, the stack properties of the new run-image MUST be compared to the stack properties of the previous run-image. If any of the following are true, the analysis MUST fail:
+* A mixin listed on the previous image does not exist on the new run-image
+* The operating system of the previous image does not matches the new run-image
+* The architecture of the previous image does not matches the new run-image
+
+During detection, the stack properties of the build image MUST be compared to the stack properties of the run-image. If any of the following are true, the analysis MUST fail:
+* The stack ID of the build-image does not matches the run-image
+* The operating system of the build-image does not matches the run-image
+* The architecture of the build-image does not matches the run-image
```

Comment:
Is this always true? I can imagine an amd64 build image that can build binaries for arm run images. Or perhaps a builder that produces scratch-like images with nothing but a launcher and the layers produced from buildpacks.

Reply:
@jkutner @sclevine would love to hear some more thoughts from others on this one.